.github/workflows/test.yaml

```diff
@@ -6,18 +6,28 @@ on:
 
 jobs:
   unit:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python_version: ['3.9', '3.10', '3.11', '3.12']
+        resolution: [lowest-direct, highest]
     steps:
     - uses: actions/checkout@v4
     - name: Install poetry
       run: pipx install poetry
     - uses: actions/setup-python@v5
       with:
-        python-version-file: pyproject.toml
-        cache: poetry
+        python-version: ${{ matrix.python_version }}
+        cache: pip
```

Comment:
```suggestion
        cache: poetry
```

Reply:
Or does this not work with the version matrix?

Reply:
Copied this from @ReinderVosDeWael implementation haha

Reply:
I think because it uses `uv pip install` it is more efficient to cache pip?

Reply:
honestly :shrug: . This is a bit of an edge case for installing that allows us to do the `lowest-direct` and `highest` resolution.